### PR TITLE
Head request

### PR DESCRIPTION
**Pull request by junglie85**
Implements HEAD requests and a large amount of refactoring to make the server simpler.

Still TODO:

1. a bunch of `if/else if/else` or a reasonably sized `switch` statement is quickly going to build up in `HttpRoute -> handleRequest(Request request)` which I still need to decide how best to deal with.
2. Decide whether the `switch` statements in the parser can be implemented another way. Suspect this is going to be the best approach and it's confined to one place, so will probably leave unless anyone can suggest a better alternative.

### CODE REVIEW

**Comment by ssunday** [comment]
I feel like the commits could be squashed together since they're basically all together head + refactor

File: src/main/java/uk/ashleybye/httpserver/http/router/HttpRoute.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      response.setProtocolVersion(request.getProtocolVersion());
+      response.setStatusCode(StatusCode.NOT_FOUND);
+      response.setBody("");
+      return response;
+    };
+  }
+
+  @Override
+  public Response handleRequest(Request request, Response response) {
+    if (request.getMethod().equals(RequestMethod.HEAD)) {
```
Comment (ssunday) [inline src/main/java/uk/ashleybye/httpserver/http/router/HttpRoute.java:31]: Would a switch statement be cleaner for this?

Comment (ssunday) [reply src/main/java/uk/ashleybye/httpserver/http/router/HttpRoute.java:31]: Esp. as you add more

Comment (junglie85) [reply src/main/java/uk/ashleybye/httpserver/http/router/HttpRoute.java:31]: Potentially. I'm undecided. I'm also wondering if I can do this via polymorphic dispatch instead and remove the need for an if or switch.

**Pull request merged**